src/main/java/VASSAL/build/module/map/KeyBufferer.java

```diff
@@ -169,9 +191,21 @@ public void mouseReleased(MouseEvent evt) {
     PieceVisitorDispatcher d = createDragSelector(
       !evt.isControlDown(), evt.isAltDown(), map.componentToMap(selection)
     );
+    
+    //BR// If it was a legit band-select drag (not just a click), our special case only applies if piece is allowed to be band-selected
+    if (bandSelectPiece != null) {
+      Point finish = map.mapToComponent(evt.getPoint());
+      EventFilter bandFilter = (EventFilter) bandSelectPiece.getProperty(Properties.BAND_SELECT_EVENT_FILTER);
+      if ((bandFilter != null) && !(evt.isAltDown() && (bandFilter instanceof Immobilized.UseAlt))) {
```

Comment:
For slightly better readability, this could be written like this (also, `Point finish` can be declared inside the next if):
```
    if (bandSelectPiece != null) {
      final EventFilter bandFilter = (EventFilter) bandSelectPiece.getProperty(Properties.BAND_SELECT_EVENT_FILTER);
      final boolean pieceAllowedToBeBandSelected = (bandFilter != null) && !(evt.isAltDown() && (bandFilter instanceof Immobilized.UseAlt));

      if (pieceAllowedToBeBandSelected) {
        final Point finish = map.mapToComponent(evt.getPoint());
        
        //BR// Open to suggestions about a better way to distinguish "click" from "lasso" (not that Vassal doesn't already suck immensely at click-vs-drag threshhold). FWIW, this "works".
        final boolean isLasso = finish.distance(anchor) >= 10;
        if (isLasso) { 
          bandSelectPiece = null;
        }
      }
    }
```

Reply:
Great - added